perk-store-services/src/main/java/org/exoplatform/perkstore/service/PerkStoreService.java

```diff
@@ -1082,6 +1082,10 @@ private boolean canViewProduct(Product product, String username, boolean isPerkS
       return true;
     }
 
+    if (username.equals(product.getCreator().getId())) {
```

Comment:
Could you add more checks on nullity for product.getCreator and move this instruction under the next one that checks the emptiness of username ?

Reply:
ok